core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -323,6 +324,18 @@
     private final WarningCollector warningCollector;
     private final CorrelationSupport correlationSupport;
 
+    private static boolean hideInaccesibleColumns;
```

Comment:
@kokosing I agree, and I must change it because it could cause some trouble, but adding it in the constructor has a huge impact because it is instantiated also by `ExpressionAnalyzer` and the value must be propagated around. This was the reason why my first choise was the session property and than put it in the `Metadata` or in the `AccessControl`. Do you haven any other suggestion?

Reply:
If you follow https://github.com/trinodb/trino/pull/9991#discussion_r758242399 then I think using session property is ok.

Reply:
ok